plugins/fontdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,514 @@
+local CenterContainer = require("ui/widget/container/centercontainer")
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local Device = require("device")
+local Font = require("ui/font")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local Menu = require("ui/widget/menu")
+local NetworkMgr = require("ui/network/manager")
+local Screen = require("device").screen
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local ltn12 = require("ltn12")
+local https = require("ssl.https")
+local util = require("util")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local featured = require("featured")
+local fontsearch = require("fontsearch")
+
+local FontDownloader = WidgetContainer:new{
+    name = "fontdownloader",
+    is_doc_only = false,
+    base_url = "https://www.googleapis.com/webfonts/v1/webfonts",
+    user_cache = DataStorage:getDataDir() .. "/cache/gbooks.lua",
+    user_key = DataStorage:getSettingsDir() .. "gfonts-api.txt",
+    timestamp_format = "%Y%m%d",
+    fonts = {},
+    blacklist = {
+        "Noto Sans", "Noto Sans HK", "Noto Sans JP", "Noto Sans KR", "Noto Sans SC", "Noto Sans TC",
+        "Noto Serif", "Noto Serif HK", "Noto Serif JP", "Noto Serif KR", "Noto Serif SC", "Noto Serif TC",
```

Comment:
We don't ship any `Noto Serif CJK` - so may be you could unblacklist `Noto Serif SC` - assuming:
- CJK users really care about Sans vs Serif :)
- Serif SC includes the TC/KR/JP variations like our NotoSansCJKsc does, which will be correctly picked up by the recent typography stuff

Reply:
> Serif SC includes the TC/KR/JP variations like our NotoSansCJKsc does, which will be correctly picked up by the recent typography stuff

Never checked but it looks like is splited in "Noto Serif HK", "Noto Serif JP", "Noto Serif KR", "Noto Serif SC" and "Noto Serif TC". So we need to unblacklist all of them.

Reply:
https://www.google.com/get/noto/help/cjk/ states that:
> Each font sets one language as the default. Note that each language-specific font **does** support all four languages and includes the complete set of glyphs

I don't know which thing your plugin will download, but you can try to see if the 5 are proposed in the list of font supporting the Korean script?
And/or download one of them (SC, for consistency), and use the `Generate font test document` and go to that font, and try to toggle `Typography rules>` between chinese and japanese: the few CJK chars in that sample should slightly change (check they do with Noto Sans CJK).

Reply:
Nope. Each noto serif fonts support two subsets: latin and its own (based on names).
![2020-06-26_16-53](https://user-images.githubusercontent.com/975883/85870646-98ebf280-b7cd-11ea-81a9-ce1033bb5d97.png)

It makes some sense as the api is intended mostly to provide font faces to webpages and there's no need to send the whole CJK when the page is written in just one language.

Other thing to consider is to release our own Noto Serif CJK somewhere and include it like the Lit3rata font.